x/finality/keeper/power_dist_change.go

```diff
@@ -346,6 +348,13 @@ func (k Keeper) ProcessAllPowerDistUpdateEvents(
 			// This is a consumer FP rather than Babylon FP, skip it
 			continue
 		}
+		// if the fp is slashed it shouldn't be included in the newDc
+		if newFP.IsSlashed() {
```

Comment:
check that fixes the issue

Reply:
I'm not sure if this fixes the issue due to the cache. Suppose `newFP` is saved when it's not slashed yet. Then at a later point it gets slashed. However, since at the time of saving `newFP` is not slashed, `newFP.IsSlashed()` will still be false. Did I miss anything?